In [ ]:
from google.colab import drive
from src.data.test_builder import process_netflix

In [ ]:
# mount to gg drive
drive.mount('/gdrive')

working_dir = '/gdrive/MyDrive/Netflix_Prize'
os.chdir(working_dir)

In [ ]:
dfs = [
    'combined_data_1.parquet',
    'combined_data_2.parquet',
    'combined_data_3.parquet',
    'combined_data_4.parquet',
]

probe_path = 'probe.txt'
save_dir = 'processed'

In [ ]:
# Process files

train_files, test_file = process_netflix(dfs, probe_path, save_dir)

print("Complete")
print(f"Train files: {train_files}")
print(f"Test file: {test_file}")

In [6]:
pd.read_parquet('processed/test.parquet')

movie_id  user_id  rating       date
0               1    30878     4.0 2005-12-26
1               1  2647871     4.0 2005-12-30
2               1  1283744     3.0 2004-04-17
3               1  2488120     5.0 2005-09-20
4               1   317050     5.0 2005-11-15
...           ...      ...     ...        ...
1408390      9997  2328701     3.0 2005-12-11
1408391      9998  1288730     3.0 2005-09-06
1408392      9998  2536567     4.0 2005-03-21
1408393      9998  1107317     5.0 2005-11-29
1408394      9999  1473765     2.0 2005-11-08

[1408395 rows x 4 columns]

In [7]:
pd.read_parquet('processed/train_part_1_group_0_movies_1-571.parquet')

user_id  rating       date  movie_id
0        1488844     3.0 2005-09-06         1
1         822109     5.0 2005-05-13         1
2         885013     4.0 2005-10-19         1
3         823519     3.0 2004-05-03         1
4         893988     3.0 2005-11-17         1
...          ...     ...        ...       ...
3039741  2570498     4.0 2004-08-09       571
3039742   616755     4.0 2003-08-21       571
3039743  2090255     4.0 2002-09-12       571
3039744   645202     2.0 2005-08-06       571
3039745   912312     5.0 2003-05-06       571

[3039746 rows x 4 columns]

 The final train set contains 99,072,112 ratings. The test set (made from Netflix's probe set) contains 1,408,395 ratings.